NFIQ2/NFIQ2Algorithm/src/prediction/RandomForestML.cpp

```diff
@@ -123,6 +123,63 @@ NFIQ2::Prediction::RandomForestML::initModule(const std::string &fileName,
 	return hash;
 }
 
+#ifdef __ANDROID__
+std::string
+NFIQ2::Prediction::RandomForestML::initModule(AAssetManager *assets,
+    const std::string &fileName, const std::string &fileHash)
+{
+	if (assets == nullptr) {
+		throw NFIQ2::Exception(NFIQ2::ErrorCode::BadArguments,
+		    "Prediction::RandomForestML::initModule: AAssetManager is null");
+	}
+	if (fileName.empty()) {
+		throw NFIQ2::Exception(NFIQ2::ErrorCode::BadArguments,
+		    "Prediction::RandomForestML::initModule: "
+		    "file name is empty");
+	}
+	if (fileHash.empty()) {
+		throw NFIQ2::Exception(NFIQ2::ErrorCode::BadArguments,
+		    "Prediction::RandomForestML::initModule: "
+		    "hash value is empty");
+	}
+	AAsset *asset = AAssetManager_open(assets, fileName.c_str(),
+	    AASSET_MODE_STREAMING);
```

Comment:
I am curious as to the performance difference when opening the Asset with [`AASSET_MODE_BUFFER`](https://developer.android.com/ndk/reference/group/asset#group___asset_1gga16af7b253440dadd46a80a4b9fddba4da40ec098f4afb7c2869fa449d3059f6bb) and reading with [`AAsset_getBuffer`](https://developer.android.com/ndk/reference/group/asset#aasset_getbuffer). Unsure if this is optimized for small or large files, but is one way to eliminate the C-string read loop. Another below.

Reply:
The docs for the [Java-equivalent value](https://developer.android.com/reference/android/content/res/AssetManager#ACCESS_BUFFER) says buffer is for small, fast reads, which our model is not. Resolving.